engines/buried/graphics.cpp

```diff
@@ -0,0 +1,884 @@
+				paletteIndex = i;
+				break;
+			}
+		}
+
+		assert(paletteIndex >= 0);
+
+		return *((const byte *)bitmap->getBasePtr(point.x - x, point.y - y)) != paletteIndex;
+	}
+}
+
+byte *GraphicsManager::createDefaultPalette() const {
+	Common::SeekableReadStream *stream = _vm->getBitmapStream(700);
+
+	if (!stream)
+		error("Couldn't find bitmap 700");
+
+	stream->skip(14);
+
+	if (stream->readUint16LE() != 8)
+		error("Trying to load palette from non-8bpp image 700");
+
+	stream->skip(16);
+
+	uint32 colorsUsed = stream->readUint32LE();
+
+	if (colorsUsed != 0 && colorsUsed != 256)
+		error("Bitmap 700 is missing a full palette");
+
+	stream->skip(4);
+	byte *palette = new byte[256 * 3];
+	byte *ptr = palette;
+
+	for (uint32 i = 0; i < 256; i++) {
+		ptr[2] = stream->readByte();
+		ptr[1] = stream->readByte();
+		ptr[0] = stream->readByte();
+		stream->readByte();
+		ptr += 3;
+	}
+
+	delete stream;
+
+	// Make sure the first entry is black and the last is white
+	palette[0 * 3]   = palette[0 * 3 + 1]   = palette[0 * 3 + 2]   = 0x00;
+	palette[255 * 3] = palette[255 * 3 + 1] = palette[255 * 3 + 2] = 0xFF;
+
+	return palette;
+}
+
+Graphics::Surface *GraphicsManager::remapPalettedFrame(const Graphics::Surface *frame, const byte *palette) {
+	// This is pretty much the same as the Cinepak one
+	// It seems to work for the one video I know that needs it (SWLOGO.BTV)
+	// TODO: Merge some of this with getColor()
+
+	byte palMap[256];
+	const byte *screenPal = getDefaultPalette();
+
+	for (int i = 0; i < 256; i++) {
+		int r = palette[i * 3];
+		int g = palette[i * 3 + 1];
+		int b = palette[i * 3 + 2];
+
+		int diff = 0x7FFFFFFF;
+		byte result = 0;
+
+		for (int j = 0; j < 256; j++) {
+			int bDiff = b - (int)screenPal[j * 3 + 2];
+			int curDiffB = diff - (bDiff * bDiff);
+
+			if (curDiffB > 0) {
+				int gDiff = g - (int)screenPal[j * 3 + 1];
+				int curDiffG = curDiffB - (gDiff * gDiff);
+
+				if (curDiffG > 0) {
+					int rDiff = r - (int)screenPal[j * 3];
+					int curDiffR = curDiffG - (rDiff * rDiff);
+
+					if (curDiffR > 0) {
+						diff -= curDiffR;
+						result = j;
+
+						if (diff == 0)
+							break;
+					}
+				}
+			}
+		}
+
+		palMap[i] = result;
+	}
+
+	Graphics::Surface *convertedSurface = new Graphics::Surface();
+	convertedSurface->create(frame->w, frame->h, frame->format);
+
+	for (int y = 0; y < frame->h; y++)
+		for (int x = 0; x < frame->w; x++)
+			*((byte *)convertedSurface->getBasePtr(x, y)) = palMap[*((const byte *)frame->getBasePtr(x, y))];
+
+	return convertedSurface;
+}
+
+Common::SeekableReadStream *GraphicsManager::findArialStream(bool bold) const {
+	Common::SeekableReadStream *stream = 0;
+
+	// Try to see if the user supplied a font
+	Common::String defaultBaseName = bold ? "arialbd.ttf" : "arial.ttf";
+	stream = SearchMan.createReadStreamForMember(defaultBaseName);
+	if (stream)
+		return stream;
+
+	// HACK: Try to load the system font
+#if defined(WIN32)
```

Comment:
Having platform-specific code here is quite ugly. Perhaps support for loading system fonts should be removed from the engine while clone2727's work for using system fonts is not merged 
(https://github.com/project-cabal/cabal/compare/b1ecacfd1d724fc9099d8ddfcf513e2f09087746...6ddf87098a26b16f7c55bf9c474cc66d212d7fb9 and a few other commits).